backend/src/main/scala/cromwell/backend/io/JobPathsWithDocker.scala

```diff
@@ -4,15 +4,21 @@ import com.typesafe.config.Config
 import cromwell.backend.{BackendJobDescriptorKey, BackendWorkflowDescriptor}
 import cromwell.core.path.{Path, PathBuilder}
 
-class JobPathsWithDocker(val jobKey: BackendJobDescriptorKey,
-                         workflowDescriptor: BackendWorkflowDescriptor,
-                         config: Config,
-                         pathBuilders: List[PathBuilder] = WorkflowPaths.DefaultPathBuilders) extends WorkflowPathsWithDocker(
-  workflowDescriptor, config, pathBuilders) with JobPaths {
+object JobPathsWithDocker {
+  def apply(jobKey: BackendJobDescriptorKey,
+  workflowDescriptor: BackendWorkflowDescriptor,
+  config: Config,
+  pathBuilders: List[PathBuilder] = WorkflowPaths.DefaultPathBuilders) = {
+    val workflowPaths = new WorkflowPathsWithDocker(workflowDescriptor, config, pathBuilders)
+    new JobPathsWithDocker(workflowPaths, jobKey)
+  }
+}
+
+case class JobPathsWithDocker(override val workflowPaths: WorkflowPathsWithDocker, jobKey: BackendJobDescriptorKey) extends JobPaths {
```

Comment:
If this is no longer meant to be called by anything other than `apply` I think it could be made package `private` (can't check that myself due to ongoing IntelliJ meltdown)